docs/architecture/adr-004-genesis-modes.md

```diff
@@ -0,0 +1,186 @@
+# ADR 4: Multiple Genesis modes
+
+## Changelog
+
+- July 26, 2019: Initial draft
+
+## Context
+
+### Problem
```

Comment:
It's not clear to me after reading this section what the actual problem is? Maybe provide an example?